src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -121,6 +121,37 @@ public static CharSeq ofAll(Iterable<? extends Character> elements) {
     }
 
     /**
+     * Returns a CharSeq containing {@code n} values of a given Function {@code f}
+     * over a range of integer values from 0 to {@code n - 1}.
+     *
+     * @param n The number of elements in the CharSeq
+     * @param f The Function computing element values
+     * @return A CharSeq consisting of elements {@code f(0),f(1), ..., f(n - 1)}
+     * @throws NullPointerException if {@code n} or {@code f} are null
+     */
+    public static CharSeq fill(Integer n, Function<Integer, Character> f) {
+        Objects.requireNonNull(n, "n is null");
+        Objects.requireNonNull(f, "f is null");
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < n; i++) {
+            sb.append(f.apply(i));
+        }
+        return new CharSeq(sb.toString());
```

Comment:
<code>return of(sb);</code>

we should return static instance <code>EMPTY</code> if <code>n == 0</code>